test/test_scandir.py

```diff
@@ -250,6 +250,32 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_walk_unicode_handling(self):
+        encoding = sys.getfilesystemencoding()
+        subdir = os.path.join(TEST_PATH, 'walk').encode(encoding)
+        dirname_encoded = u'\u0064\u0061\u0064\u0061'.encode(encoding)
```

Comment:
I would prefer to have one variable for the unicode version and then encode it:

dirname_unicode = u'test_unicode_dir'
dirname_bytes = dirname_unicode.encode(encoding)